slack_notification/slack_notification.php

```diff
@@ -1,183 +1,183 @@
+            new Slack_Text("*Commit:* {$hash}"),
+            new Slack_Text("*Committed by:* {$committer}"),
+        ]);
+        $blocks[] = new Slack_Simple_Block(new Slack_Text("*Commit Message:*\n{$message}"));
+        break;
+
+    case 'clear_cache':
+        $blocks[] = new Slack_Simple_Block(new Slack_Text("*Action:*\nCaches cleared on <http://{$environment}-{$site_name}.pantheonsite.io|{$environment}>."));
+        break;
+
+    case 'clone_database':
+        $blocks[] = new Slack_Multi_Block([
+            new Slack_Text("*Cloned from:* {$_POST['from_environment']}"),
+            new Slack_Text("*Cloned to:* {$environment}"),
+        ]);
+        break;
+
+    default:
+        $description = $_POST['qs_description'] ?? 'No additional details provided.';
+        $blocks[] = new Slack_Simple_Block(new Slack_Text("*Description:*\n{$description}"));
+        break;
 }
 
+// Add a divider block at the end of the message
+$blocks[] = new Slack_Divider_Block();
+
+// Prepare attachments with yellow sidebar
+$attachments = [
+    [
+        'color' => $pantheon_yellow,
+        'blocks' => $blocks,
+    ],
+];
+
+// echo "Blocks:\n";
+// print_r( $blocks ); // DEBUG
+
+// Send the Slack notification
+_post_to_slack($attachments);
+
 /**
- * Send a notification to slack
+ * Send a notification to Slack
+ *
+ * @param array $attachments The array of attachments to include in the Slack message.
  */
-function _slack_notification($slack_url, $channel, $username, $text, $attachment, $alwaysShowText = false)
-{
-  $attachment['fallback'] = $text;
-  $post = array(
-    'username' => $username,
-    'channel' => $channel,
-    'icon_emoji' => ':lightning_cloud:',
-    'attachments' => array($attachment)
-  );
-  if ($alwaysShowText) {
-    $post['text'] = $text;
-  }
-  $payload = json_encode($post);
-  $ch = curl_init();
-  curl_setopt($ch, CURLOPT_URL, $slack_url);
-  curl_setopt($ch, CURLOPT_POST, 1);
-  curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
-  curl_setopt($ch, CURLOPT_TIMEOUT, 5);
-  curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json'));
-  curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
-  // Watch for messages with `terminus workflows watch --site=SITENAME`
-  print("\n==== Posting to Slack ====\n");
-  $result = curl_exec($ch);
-  print("RESULT: $result");
-  // $payload_pretty = json_encode($post,JSON_PRETTY_PRINT); // Uncomment to debug JSON
-  // print("JSON: $payload_pretty"); // Uncomment to Debug JSON
-  print("\n===== Post Complete! =====\n");
-  curl_close($ch);
+function _post_to_slack($attachments) {
+    // Uncomment the following line to debug the attachments array.
+    // echo "Attachments - Raw:\n"; print_r( $attachments ); echo "\n";
+
+    $slack_token = pantheon_get_secret('slack_deploybot_token'); // Set the token name to match the secret you added to Pantheon.
+
+    $post = [
+        'channel' => '#firehose', // The Slack channel to post to.
```

Comment:
There is also an unused duplicate $slack_channel variable above. I wonder if there's a good way to parameterize this. Too bad the notion of parameters in the pantheon.yml Quicksilver definition was never introduced.

Reply:
Oh good catch. That's likely from copy/pasta'd code. My intent was to use the variable.